autolens/plot/get_visuals.py

```diff
@@ -0,0 +1,172 @@
+            are extracted via the `GetVisuals2D` object).
+        visuals
+            The pre-existing visuals of the plotter which new visuals are added too via the `GetVisuals2D` class.
+        """
+        super().__init__(include=include, visuals=visuals)
+
+    def via_tracer_from(
+        self, tracer: Tracer, grid: aa.type.Grid2DLike, plane_index: int
+    ) -> aplt.Visuals2D:
+        """
+        From a `Tracer` get the attributes that can be plotted and returns them in a `Visuals2D` object.
+
+        Only attributes with `True` entries in the `Include` object are extracted.
+
+        From a tracer the following attributes can be extracted for plotting:
+
+        - origin: the (y,x) origin of the coordinate system used to plot the light object's quantities in 2D.
+        - border: the border of the mask of the grid used to plot the light object's quantities in 2D.
+        - light profile centres: the (y,x) centre of every `LightProfile` in the object.
+        - mass profile centres: the (y,x) centre of every `MassProfile` in the object.
+        - critical curves: the critical curves of all of the tracer's mass profiles combined.
+        - caustics: the caustics of all of the tracer's mass profiles combined.
+
+        When plotting a `Tracer` it is common for plots to only display quantities corresponding to one plane at a time
+        (e.g. the convergence in the image plane, the source in the source plane). Therefore, quantities are only
+        extracted from one plane, specified by the  input `plane_index`.
+
+        Parameters
+        ----------
+        tracer
+            The `Tracer` object which has attributes extracted for plotting.
+        grid
+            The 2D grid of (y,x) coordinates used to plot the tracer's quantities in 2D.
+        plane_index
+            The index of the plane in the tracer which is used to extract quantities, as only one plane is plotted
+            at a time.
+
+        Returns
+        -------
+        vis.Visuals2D
+            A collection of attributes that can be plotted by a `Plotter` object.
+        """
+        origin = self.get("origin", value=aa.Grid2DIrregular(grid=[grid.origin]))
+
+        border = self.get("border", value=grid.mask.border_grid_sub_1.binned)
+
+        if border is not None:
+            if plane_index > 0:
+                border = tracer.traced_grids_of_planes_from(grid=border)[plane_index]
+
+        light_profile_centres = self.get(
+            "light_profile_centres",
+            tracer.planes[plane_index].extract_attribute(
+                cls=ag.lp.LightProfile, attr_name="centre"
+            ),
+        )
+
+        mass_profile_centres = self.get(
+            "mass_profile_centres",
+            tracer.planes[plane_index].extract_attribute(
+                cls=ag.mp.MassProfile, attr_name="centre"
+            ),
+        )
+
+        if plane_index == 0:

```

Comment:
Could be a usecase for the factory pattern where you would create a different instance in the `__new__` method depending on the plane index thereby avoiding checking its value